src/app/services/auth.service.ts

```diff
@@ -57,21 +58,43 @@ export class AuthService {
   }
 
 
-  // isTokenExpired(token: string): boolean {
-  //   const decodedToken: any = jwtDecode(token)
-  //   return (decodedToken.exp * 1000) < Date.now()
-  // }
-
   private handleError(error: HttpErrorResponse) {
-    let errorMessage = 'An unknown error occurred';
+    let errorMessage: string;
+
     if (error.error instanceof ErrorEvent) {
-      // Client-side or network error
+    
       errorMessage = `Error: ${error.error.message}`;
     } else {
-      // Backend error
-      errorMessage = `Error Code: ${error.status}\nMessage: ${error.message}`;
+    
+      switch (error.status) {
+        case 400:
+
+        errorMessage = `Bad Request: ${error.error?.message || error.message}`   
+          break;
+
+          case 401: 
+        errorMessage  = 'unathorized: You need to log in to access this resource';  
+          break;
+
+          case 500: 
+        errorMessage = 'Internal Server Error: Please try again later';
+          break;
+          
+          case 503:
+        errorMessage = 'Sevice Unavailable: The server is temporarily unable to handle the request';
+           break;    
+
+        default:
+          errorMessage = `Error Code: ${error.status}\nMessage: ${error.error?.message || error.message || 'Unknown error'}
+          }`
+      }
     }
-    console.error(errorMessage);
+
+    // log the error for debugging
+    console.error('Error Details: ', {
```

Comment:
instead of this console.error, is it possible to to toast the message to the user. Explore the option of using a toaster

Reply:
done now using ngx-toastr to handle all notification messages